lib/parser/parser_stream.js

```diff
@@ -28,12 +29,16 @@ inherits(ParserStream, WritableStream);
 
 //WritableStream implementation
 ParserStream.prototype._write = function (chunk, encoding, callback) {
+    this.written = true;
     this.writeCallback = callback;
     this.parser.tokenizer.write(chunk.toString('utf8'), this.lastChunkWritten);
     this._runParsingLoop();
 };
 
 ParserStream.prototype.end = function (chunk, encoding, callback) {
+    if (!this.written && typeof chunk === 'undefined')
```

Comment:
BTW, I believe it's always safe to write empty string here, so I guess the best way will be to eliminate `this.written` variable and use the second code snippet suggested by Ingvar.